### DISCUSSION TITLE
Using CAN-FD with differing speeds

### DISCUSSION THREAD

**Discussion by tyalie** [collin80/SavvyCAN#684] [Q&A]
Is it possible to use SavvyCAN CAN-FD with different speeds for CAN and CAN-FD?

**Comment by collin80** [comment]
Well, there is no way to do that in the program right now. CAN-FD support is still extremely limited and so there isn't a way to set the speed for CAN-FD. But, there are a couple of options that could still work:

1. Things that use the GVRET protocol (like the EVTV dongles I made) can be set from a serial terminal and then SavvyCAN will just use whatever you had set in the terminal so you can set CAN-FD speed that way

2. Likewise, any socketcan device in linux is set from the command line so you can just set the parameters there and then tell SavvyCAN to connect to the appropriate socketcan interface.

**Reply by tyalie** [reply]
Yeah. I've now choosen the brutal route and have hardcoded the bitrate and data bitrate in the program. But it doesn't seem to work on Windows rn

**Reply by walnut-engineering** [reply]
I was looking to go down this route as I want to use 500k/2M.  Are you using a peak device?

**Reply by tyalie** [reply]
Yes. I've just done this here and hardcoded the required speed. QT actually exposes the interface. But note that 1M CAN-FD doesn't work due to restrictions set by the QT library.

https://github.com/EXOBOS/SavvyCAN/commit/22338d1e3c4b2573816856b84f9f37e2e37cff45

The repo has a really nice CD/CI pipeline so you don't need to compile it on your desktop you just need to enable GitHub actions after forking.

**Comment by danjhunter** [comment]
@collin80 How might I set EVTV CanDuoV2.0 from serial terminal? Could I just serial connect to the Duo board via Coolterm at 115200 baud rate and type a command to set the board to 2Mbit/sec CanFD?

**Reply by collin80** [reply]
If you use coolterm then tell it the baud rate is 1000000 (1 million). Then you can issue commands and see replies

**Reply by danjhunter** [reply]
Thanks @collin80. I am able to see the CANFD stream via serial at 2Mbit/sec with cool term now! Data looks like : 427557048 - 40023 X 1 1 0
**!!****!!**!!**!!**!!**!!****!!********!!**!!!!!!**!!!!**!!******!!**!!**!!!!**!!****!!!!!!**!!****!!******!!**!!!!!!!!**!!**!!****!!**!!****!!**!!!!**!!**!!**!!******!!**!!!!**!!**!!**!!**!!****427635049 - 4000a X 1 1 0. Any ideas why SavvyCAN is unable to load this data stream? I'm currently stuck streaming via CoolTerm. Any advice or pointers would be helpful.

**Comment by danjhunter** [comment]
Connected to the CANdueV2 I tried to send the following hex in the SavvyCAN Connection Console:  `f1 5 20 a1 7 c0 80 84 1e c0 0`. This should change the baud rate on channel 0 to 500kbit/sec and channel 1 to 2Mbit/sec. (given little endian and 3 byte hex 80 84 1e is 2,000,000).  It seems to accept this but then I check it I get this `f1 06 01 20 a1 07 00 01 80 84 0e 00` which means 500K // 951,424...
